client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/screen-confirmation.tsx

```diff
@@ -1,40 +1,100 @@
 import { Button } from '@automattic/components';
+import { Design } from '@automattic/design-picker';
+import { localizeUrl } from '@automattic/i18n-utils';
 import { NavigatorHeader } from '@automattic/onboarding';
 import {
 	__experimentalVStack as VStack,
 	__experimentalHStack as HStack,
 } from '@wordpress/components';
-import { Icon, image, verse, layout } from '@wordpress/icons';
+import { Icon, image, layout, home, verse } from '@wordpress/icons';
 import { useTranslate } from 'i18n-calypso';
-import { useScreen } from './hooks';
+import { useSelector } from 'react-redux';
+import QueryActiveTheme from 'calypso/components/data/query-active-theme';
+import { getActiveTheme, getCanonicalTheme } from 'calypso/state/themes/selectors';
+import { IAppState } from 'calypso/state/types';
 import NavigatorTitle from './navigator-title';
 import './screen-confirmation.scss';
 
 interface Props {
+	isNewSite: boolean;
+	siteId?: number;
+	selectedDesign?: Design;
 	onConfirm: () => void;
 }
 
-const ScreenConfirmation = ( { onConfirm }: Props ) => {
+const ScreenConfirmation = ( { isNewSite, siteId = 0, selectedDesign, onConfirm }: Props ) => {
 	const translate = useTranslate();
-	const { title, description, continueLabel } = useScreen( 'confirmation' );
 
-	const list = [
-		{
-			icon: image,
-			title: translate( 'Upload images' ),
-			description: translate( 'Showcase your photos in their best light.' ),
-		},
-		{
-			icon: verse,
-			title: translate( 'Start writing' ),
-			description: translate( 'Get things going and share your insights.' ),
-		},
-		{
-			icon: layout,
-			title: translate( 'Customize every detail' ),
-			description: translate( 'Make your site even more unique.' ),
-		},
-	];
+	const currentThemeId = useSelector( ( state: IAppState ) => getActiveTheme( state, siteId ) );
+	const currentTheme = useSelector( ( state: IAppState ) =>
+		getCanonicalTheme( state, siteId, currentThemeId )
+	);
+	const willThemeChange = currentThemeId !== selectedDesign?.slug;
```

Comment:
It has a chance the value is true initially and then becomes false if we never fetch the current active theme somewhere.

https://github.com/Automattic/wp-calypso/assets/13596067/3b0978b7-d9ac-49f1-96ca-3029447b6069

Reply:
Thanks for catching; I've fixed that in the last commit (https://github.com/Automattic/wp-calypso/pull/85410/commits/491ac743bcf6ddb843a99a3932623d524db8d7b2) although not really elegant 🥲

Reply:
It looks okay for now 👍